### PR TITLE
Production Release

### PR DESCRIPTION
**Pull request by tomahawk-pilot**
Production Release July 12, 2020
Changelog:

1) Hard refresh button
2) changes to the patient profile
3) changes to sample workflow
4) Various CSV download options
5) Various other fixes

Contributors:
**1. @MalayaliRobz 
2. @aashnanaushad 
3. @bodhisha 
4. @psjishnu 
5. @tomahawk-pilot**

### CODE REVIEW

File: src/Components/Patient/SampleTest.tsx [hunk 1]
```diff
@@ -82,19 +120,19 @@ export const SampleTest = (props: any) => {
           }
           break;
         case "icmr_category":
-          if ( !state.form[field]) {
+          if (!state.form[field]) {
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:123]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

**Comment by tomahawk-pilot** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 19
- Added 52
           

Complexity increasing per file
==============================
- build-scripts/generate-build-version.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477&bid=19115919)

File: src/Components/Patient/SampleTest.tsx [hunk 2]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
+          ? state.form.atypical_presentation
+          : undefined,
         diagnosis: state.form.diagnosis ? state.form.diagnosis : undefined,
-        diff_diagnosis: state.form.diff_diagnosis ? state.form.diff_diagnosis : undefined,
-        doctor_name: state.form.doctor_name ? state.form.doctor_name : undefined,
-        etiology_identified: state.form.etiology_identified ? state.form.etiology_identified : undefined,
+        diff_diagnosis: state.form.diff_diagnosis
+          ? state.form.diff_diagnosis
+          : undefined,
+        testing_facility: state.form.testing_facility,
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:185]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Facility/FacilityCreate.tsx [hunk 3]
```diff
@@ -237,7 +237,7 @@ export const FacilityCreate = (props: FacilityProps) => {
           return;
         case "phone_number":
           const phoneNumber = parsePhoneNumberFromString(state.form[field]);
-          if (!state.form[field] || !phoneNumber?.isPossible()) {
+          if (!state.form[field] || !phoneNumber?.isPossible() || !phonePreg(String(phoneNumber?.number))) {
```
Comment (tomahawk-pilot) [inline src/Components/Facility/FacilityCreate.tsx:240]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 4]
```diff
@@ -182,14 +181,39 @@ export default function UserProfile() {
...
     };
 
-    return (
+    const checkForNewBuildVersion = async () => {
+
+        let [isLatestBundle, newVersion] = await checkIfLatestBundle();
+
+        if (!isLatestBundle) {
+            setUpdateBtnText("updating...");
+            localStorage.setItem('build_meta_version', newVersion);
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:196]: Codacy found an issue: [localStorage can be unsafe](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+const jsonData = {
+  version: appVersion
+};
+
+const jsonContent = JSON.stringify(jsonData);
+
+fs.writeFile(
+  './public/build-meta.json',
+  jsonContent,
+  { flag: 'w+', encoding: 'utf8' },
```
Comment (tomahawk-pilot) [inline build-scripts/generate-build-version.js:17]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/models.tsx [hunk 6]
```diff
@@ -1,188 +1,205 @@
...
+  allow_transfer?: boolean;
+  discharge?: boolean;
+  gender?: number;
+  phone_number?: string;
+  allergies?: string;
+  medical_history?: Array<{ disease: string | number; details: string }>;
+  facility_object?: {
+    id: number;
+    name: string;
+    facility_type?: { id: number; name: string };
```
Comment (tomahawk-pilot) [inline src/Components/Patient/models.tsx:29]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 7]
```diff
@@ -165,14 +166,12 @@ export default function UserProfile() {
             setIsLoading(true);
             const data = {
                 username: username,
-                first_name: states.form.first_name,
-                last_name: states.form.last_name,
+                first_name: states.form.firstName,
+                last_name: states.form.lastName,
                 email: states.form.email,
-                phone_number: parsePhoneNumberFromString(states.form.phone_number)?.format('E.164'),
+                phone_number: parsePhoneNumberFromString(states.form.phoneNumber)?.format('E.164'),
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:172]: Codacy found an issue: [Identifier 'phone_number' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 8]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
+        doctor_name: state.form.doctor_name
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:186]: Codacy found an issue: [Identifier 'doctor_name' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 9]
```diff
@@ -66,10 +75,39 @@ export const SampleTest = (props: any) => {
...
+      const FacilityNameList = await dispatchAction(
+        getAllFacilities({ facility_type })
+      );
+      if (!status.aborted && FacilityNameList.data.results) {
+        setFacilityName([...FacilityNameList.data.results]);
+        dispatch({
+          type: "set_form",
+          form: {
+            ...state.form,
+            testing_facility: FacilityNameList.data.results[0]?.id,
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:97]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 10]
```diff
@@ -127,9 +128,9 @@ export default function UserProfile() {
                         invalidForm = true;
                     }
                     return;
-                case "phone_number":
-                    const phoneNumber = parsePhoneNumberFromString(states.form[field]);
-                    if (!states.form[field] || !phoneNumber?.isPossible()) {
+                case "phoneNumber":
+                    const phoneNumber = parsePhoneNumberFromString(states.form[field])?.number;
+                    if (!states.form[field] || !phonePreg(String(phoneNumber))) {
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:133]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 11]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
         has_sari: state.form.has_sari,
         is_unusual_course: state.form.is_unusual_course,
         is_atypical_presentation: state.form.is_atypical_presentation,
-        atypical_presentation: state.form.is_atypical_presentation ? state.form.atypical_presentation : undefined,
+        atypical_presentation: state.form.is_atypical_presentation
+          ? state.form.atypical_presentation
+          : undefined,
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:180]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 12]
```diff
@@ -66,10 +75,39 @@ export const SampleTest = (props: any) => {
...
+  const [facilityName, setFacilityName] = useState<Array<FacilityNameModel>>(
+    []
+  );
 
   const headerText = "Request Sample";
   const buttonText = "Confirm your request to send sample for testing";
 
+  const fetchFacilityName = useCallback(
+    async (status: statusType) => {
+      const facility_type = 950;
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:87]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 13]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
-        doctor_name: state.form.doctor_name ? state.form.doctor_name : undefined,
-        etiology_identified: state.form.etiology_identified ? state.form.etiology_identified : undefined,
+        diff_diagnosis: state.form.diff_diagnosis
+          ? state.form.diff_diagnosis
+          : undefined,
+        testing_facility: state.form.testing_facility,
+        doctor_name: state.form.doctor_name
+          ? state.form.doctor_name
+          : undefined,
+        etiology_identified: state.form.etiology_identified
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:189]: Codacy found an issue: [Identifier 'etiology_identified' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const { v4: uuidv4 } = require('uuid');
```
Comment (tomahawk-pilot) [inline build-scripts/generate-build-version.js:4]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 15]
```diff
@@ -182,14 +181,39 @@ export default function UserProfile() {
                 Notification.Success({
                     msg: "Details updated successfully"
                 });
-                window.location.reload();
+                setDetails({ ...details, first_name: states.form.firstName, last_name: states.form.lastName, age: states.form.age, gender: genderTypes.filter(el => { return el.id === Number(states.form.gender) })[0].text, email: states.form.email, phone_number: states.form.phoneNumber });
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:184]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:184]: Codacy found an issue: [Identifier 'first_name' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 16]
```diff
@@ -77,12 +78,12 @@ export default function UserProfile() {
                 if (res && res.data) {
                     setDetails(res.data);
                     const formData: any = {
-                        first_name: res.data.first_name,
-                        last_name: res.data.last_name,
+                        firstName: res.data.first_name,
+                        lastName: res.data.last_name,
                         age: res.data.age,
-                        gender: res.data.gender,
+                        gender: genderTypes.filter(el => { return el.text === res.data.gender })[0].id,
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:84]: Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 17]
```diff
@@ -165,14 +166,12 @@ export default function UserProfile() {
             setIsLoading(true);
             const data = {
                 username: username,
-                first_name: states.form.first_name,
-                last_name: states.form.last_name,
+                first_name: states.form.firstName,
+                last_name: states.form.lastName,
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:170]: Codacy found an issue: [Identifier 'last_name' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 18]
```diff
@@ -7,17 +7,17 @@ import { useDispatch, useSelector } from "react-redux";
 import { getUserDetails, updateUserDetails } from "../../Redux/actions";
 import { PhoneNumberField, SelectField, TextInputField } from "../Common/HelperInputFields";
 import { parsePhoneNumberFromString } from 'libphonenumber-js';
-import { validateEmailAddress } from "../../Common/validation";
+import { validateEmailAddress, phonePreg } from "../../Common/validation";
 import * as Notification from "../../Utils/Notifications.js";
-
+import { checkIfLatestBundle } from '../../Utils/build-meta-info';
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:12]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Common/constants.tsx [hunk 19]
```diff
@@ -45,18 +52,22 @@ export const FACILITY_TYPES: Array<OptionsType> = [
...
+  { id: 802, text: "Family Health Centres" },
+  { id: 803, text: "Community Health Centres" },
+  { id: 820, text: "Urban Primary Health Centre" },
+  { id: 830, text: "Taluk Hospitals" },
+  { id: 831, text: "Taluk Headquarters Hospitals" },
+  { id: 840, text: "Women and Child Health Centres" },
+  { id: 850, text: "General hospitals" },
+  { id: 860, text: "District Hospitals" },
+  { id: 870, text: "Govt Medical College Hospitals"},
+  { id: 950, text: 'Corona Testing Labs'},
```
Comment (tomahawk-pilot) [inline src/Common/constants.tsx:68]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Facility/HospitalList.tsx [hunk 20]
```diff
@@ -62,21 +126,25 @@ export const HospitalList = () => {
   };
   const onSearchSuspects = async (searchValue: string) => {
     setIsLoading(true);
-    const res = await dispatchAction(getFacilities({ limit, offset, search_text: searchValue }));
+    const res = await dispatchAction(
+      getFacilities({ limit, offset, search_text: searchValue })
```
Comment (tomahawk-pilot) [inline src/Components/Facility/HospitalList.tsx:130]: Codacy found an issue: [Identifier 'search_text' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 21]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
         has_sari: state.form.has_sari,
         is_unusual_course: state.form.is_unusual_course,
         is_atypical_presentation: state.form.is_atypical_presentation,
-        atypical_presentation: state.form.is_atypical_presentation ? state.form.atypical_presentation : undefined,
+        atypical_presentation: state.form.is_atypical_presentation
```
Comment (tomahawk-pilot) [inline src/Components/Patient/SampleTest.tsx:178]: Codacy found an issue: [Identifier 'atypical_presentation' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Redux/actions.tsx [hunk 22]
```diff
@@ -38,9 +38,27 @@ export const getFacilities = (params: object) => {
 export const getFacility = (id: number) => {
   return fireRequest('getFacility', [id], {});
 };
+export const getAllFacilities = (params: object) => {
+  return fireRequest('getAllFacilitiesList', [], params);
```
Comment (tomahawk-pilot) [inline src/Redux/actions.tsx:42]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 23]
```diff
@@ -257,6 +281,28 @@ export default function UserProfile() {
...
+                                            Not Verified
+                                        </dd>
+                                        }
+                                    </div>
+                                    <div className="sm:col-span-1">
+                                        <dt className="text-sm leading-5 font-medium text-gray-500">
+                                            Access Level
+                                        </dt>
+                                        <dd className="mt-1 badge badge-pill badge-primary text-sm leading-5 text-gray-900">
+                                            {details.user_type || '-'}
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:303]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 24]
```diff
@@ -182,14 +181,39 @@ export default function UserProfile() {
...
+                    for (const name of names) {
+                        caches.delete(name);
+                    }
 
-        <div >
+                    window.location.reload(true);
+                });
+            }
+        } else {
+            setUpdateBtnText("You already have the latest version!")
```
Comment (tomahawk-pilot) [inline src/Components/Users/UserProfile.tsx:209]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

**Comment by dauntlessnomad** [comment]
Depends on https://github.com/coronasafe/care/pull/425

**Comment by dauntlessnomad** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 36
- Added 50
           

Complexity increasing per file
==============================
- build-scripts/generate-build-version.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477&bid=19115919)

File: src/Components/Patient/SampleTest.tsx (hunk 2 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:185]: Codacy found an issue: [Identifier 'testing_facility' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx (hunk 23 above)
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:303]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 25]
```diff
@@ -356,11 +402,11 @@ export default function UserProfile() {
                                             <div className="col-span-6 sm:col-span-3">
                                                 <PhoneNumberField
                                                     label="Phone Number*"
-                                                    value={states.form.phone_number}
+                                                    value={states.form.phoneNumber}
                                                     onChange={(value: any) => [
-                                                        handleValueChange(value, 'phone_number')
+                                                        handleValueChange(value, 'phoneNumber')
```
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:407]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+
+const jsonContent = JSON.stringify(jsonData);
+
+fs.writeFile(
+  './public/build-meta.json',
+  jsonContent,
+  { flag: 'w+', encoding: 'utf8' },
+  err => {
+    if (err) {
+      console.log('An error occured while writing JSON Object to meta.json');
```
Comment (dauntlessnomad) [inline build-scripts/generate-build-version.js:20]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Facility/HospitalList.tsx (hunk 20 above)
Comment (dauntlessnomad) [inline src/Components/Facility/HospitalList.tsx:130]: Codacy found an issue: [Identifier 'search_text' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx (hunk 15 above)
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:184]: Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js [hunk 27]
```diff
@@ -0,0 +1,26 @@
...
+      return console.log(err);
+    }
+    console.log('build-meta.json file has been saved with latest version number');
```
Comment (dauntlessnomad) [inline build-scripts/generate-build-version.js:23]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Common/constants.tsx [hunk 28]
```diff
@@ -45,18 +52,22 @@ export const FACILITY_TYPES: Array<OptionsType> = [
...
-  { id: 920, text: "Urban Primary Health Centre" },
-  { id: 930, text: "Taluk Hospitals" },
-  { id: 931, text: "Taluk Headquarters Hospitals" },
-  { id: 940, text: "Women and Child Health Centres" },
-  { id: 950, text: "General hospitals" },
-  { id: 960, text: "District Hospitals" },
-  {id: 1000, text: "Corona Care Centre"},
-  {id: 1100, text: "First Line Treatment Centre"}
+  { id: 8, text: 'Govt Hospital'},
+  { id: 9, text: 'Labs'},
```
Comment (dauntlessnomad) [inline src/Common/constants.tsx:56]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/models.tsx (hunk 6 above)
Comment (dauntlessnomad) [inline src/Components/Patient/models.tsx:29]: Codacy found an issue: [Identifier 'facility_type' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx (hunk 4 above)
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:196]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js (hunk 26 above)
Comment (dauntlessnomad) [inline build-scripts/generate-build-version.js:20]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx (hunk 16 above)
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:84]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 29]
```diff
@@ -182,14 +181,39 @@ export default function UserProfile() {
...
+                    }
 
-        <div >
+                    window.location.reload(true);
+                });
+            }
+        } else {
+            setUpdateBtnText("You already have the latest version!")
+
+            setTimeout(() => setUpdateBtnText("Update"), 1000);
```
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:211]: Codacy found an issue: [The function setTimeout can be unsafe](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Redux/actions.tsx (hunk 22 above)
Comment (dauntlessnomad) [inline src/Redux/actions.tsx:42]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js [hunk 30]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
```
Comment (dauntlessnomad) [inline build-scripts/generate-build-version.js:3]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js [hunk 31]
```diff
@@ -0,0 +1,26 @@
...
+const appVersion = uuidv4();
+
+const jsonData = {
+  version: appVersion
+};
+
+const jsonContent = JSON.stringify(jsonData);
+
+fs.writeFile(
+  './public/build-meta.json',
```
Comment (dauntlessnomad) [inline build-scripts/generate-build-version.js:15]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx [hunk 32]
```diff
@@ -182,14 +181,39 @@ export default function UserProfile() {
...
-    return (
+    const checkForNewBuildVersion = async () => {
+
+        let [isLatestBundle, newVersion] = await checkIfLatestBundle();
+
+        if (!isLatestBundle) {
+            setUpdateBtnText("updating...");
+            localStorage.setItem('build_meta_version', newVersion);
+
+            if ('caches' in window) {
```
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:198]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx (hunk 17 above)
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:170]: Codacy found an issue: [Identifier 'last_name' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Users/UserProfile.tsx (hunk 10 above)
Comment (dauntlessnomad) [inline src/Components/Users/UserProfile.tsx:133]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js (hunk 5 above)
Comment (dauntlessnomad) [inline build-scripts/generate-build-version.js:17]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 33]
```diff
@@ -131,21 +175,33 @@ export const SampleTest = (props: any) => {
...
+        doctor_name: state.form.doctor_name
+          ? state.form.doctor_name
+          : undefined,
+        etiology_identified: state.form.etiology_identified
+          ? state.form.etiology_identified
+          : undefined,
         sample_type: state.form.sample_type,
         icmr_category: state.form.icmr_category,
-        sample_type_other: state.form.sample_type === 'OTHER TYPE' ? state.form.sample_type_other : undefined,
+        sample_type_other:
```
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:194]: Codacy found an issue: [Identifier 'sample_type_other' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx (hunk 13 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:189]: Codacy found an issue: [Identifier 'etiology_identified' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx (hunk 11 above)
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:180]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: build-scripts/generate-build-version.js [hunk 34]
```diff
@@ -0,0 +1,26 @@
...
+  jsonContent,
+  { flag: 'w+', encoding: 'utf8' },
+  err => {
```
Comment (dauntlessnomad) [inline build-scripts/generate-build-version.js:18]: Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

File: src/Components/Patient/SampleTest.tsx [hunk 35]
```diff
@@ -105,6 +143,12 @@ export const SampleTest = (props: any) => {
             invalidForm = true;
           }
           break;
+        case "testing_facility":
+          if (!state.form[field]) {
+            errors[field] = "Please Choose a testing facility";
```
Comment (dauntlessnomad) [inline src/Components/Patient/SampleTest.tsx:148]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)

**Pull request merged**